src/model_filereading.cpp

```diff
@@ -274,23 +284,25 @@ bool isAtomLine(const std::vector<std::string>& substrings) {
 
 // reads a string and converts it to valid atom symbol: first character uppercase followed by lowercase characters
 std::string strToValidSymbol(std::string str){
+  // return empty if str is empty
+  if (str.size() == 0){return "";}
+  else{
+    // return empty if str begins with non-alphabetic character
+    if (!isalpha(str[0])){return "";}
+    // only for first character in sequence, convert to uppercase
+    else {str[0] = toupper(str[0]);}
+  }
   // iterate over all characters in string
-  for (size_t i = 0; i<str.size(); i++) {
-// TODO: decide whether non-alphabetic characters should be erased or not
-//    if (isalpha(str[i])){
-      // only for first character in sequence, convert to uppercase
-      if (i==0) {
-        str[i] = toupper(str[i]);
-      }
-      // for all other characters, convert to lowercase
-      else {
-        str[i] = tolower(str[i]);
-      }
-// TODO bis
-//    }
-//    else { // Remove number or charges from atoms so that "Pd2+" becomes "Pd"
-//      str.erase(i, str.size());
-//    }
+  for (size_t i = 1; i < str.size(); i++) {
+    // convert to lowercase
+    if (isalpha(str[i])) {
+      str[i] = tolower(str[i]);
+    }
+    // allow underscores inside element symbols for custom symbols
+    else if (str[i] == '_') {}

```

Comment:
Underscores don't show properly in unicode chemical formula because it was not considered as a potential character. We should either remove underscore or adapt unicode chemical formula conversion function.

Reply:
I'm removing underscores for now. If there's a need it's simple to add them back in and adapt the conversion function.